client/cmd/systemd/systemd.go

```diff
@@ -0,0 +1,176 @@
+package systemd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+type runner interface {
+	Run(name string, args ...string) (string, error)
+}
+
+type realRunner struct{}
+
+func (realRunner) Run(name string, args ...string) (string, error) { return run(name, args...) }
+
+var (
+	execRunner    runner = realRunner{}
+	isLinux              = func() bool { return runtime.GOOS == "linux" }
+	lookupSystemd        = func() error {
+		_, err := exec.LookPath("systemctl")
+		return err
+	}
+)
+
+type Options struct {
+	ServiceName string
+	ExecPath    string
+	ExecArgs    string
+	Env         map[string]string
+	UserMode    bool
+	UnitPath    string
+	WantedBy    string
+}
+
+func Install(opts Options) error {
```

Comment:
another point to have in mind is whether we want this to be installed and start, or if the user will have to run
`hoop install systemd agent` and then `hoop start systemd` 

Splitting this into two commands makes each command have only one responsibility. On the other hand, this will add the complexity of running two commands, which is not too bad